### ISSUE TITLE
CI: Suite-k8s-1.16.K8sUpdates Tests upgrade and downgrade from a Cilium stable image to master

### ISSUE THREAD

**Issue by sayboras** [cilium/cilium#23845]
### Stacktrace
```
/home/jenkins/workspace/Cilium-PR-K8s-1.16-kernel-4.19/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:515
DNS entry is not ready after timeout
Expected
    <*errors.errorString | 0xc0006f6fb0>: {
        s: "DNS 'app1-service.default.svc.cluster.local' is not ready after timeout: 7m0s timeout expired",
    }
to be nil
/home/jenkins/workspace/Cilium-PR-K8s-1.16-kernel-4.19/src/github.com/cilium/cilium/test/k8s/updates.go:374
```

Jenkins: https://jenkins.cilium.io/job/Cilium-PR-K8s-1.16-kernel-4.19/3/testReport/junit/Suite-k8s-1/16/K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master/

Sysdump: 
[2945a7bb_K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master (1).zip](https://github.com/cilium/cilium/files/10766360/2945a7bb_K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master.1.zip)

**Comment by christarazi** [comment]
Another instance here: https://jenkins.cilium.io/job/Cilium-PR-K8s-1.16-kernel-4.19/21/

[bba8e18a_K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master.zip](https://github.com/cilium/cilium/files/10771644/bba8e18a_K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master.zip)

**Comment by thorn3r** [comment]
another one here: https://github.com/cilium/cilium/pull/23772

jenkins: https://jenkins.cilium.io/job/Cilium-PR-K8s-1.16-kernel-4.19/115/testReport/Suite-k8s-1/16/K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master/

[309e31eb_K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master.zip](https://github.com/cilium/cilium/files/10796581/309e31eb_K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master.zip)

**Comment by pchaigno** [comment]
These failures increased significantly on master in recent days:
![image](https://user-images.githubusercontent.com/1764210/220457482-296ed955-8fb4-4b18-a5c5-4af72efd625d.png)
from https://lookerstudio.google.com/s/sSftc2S39TA

**Comment by pchaigno** [comment]
https://github.com/cilium/cilium/pull/23343 seems like a likely culprit given it significantly changed the Cilium configuration we run on 4.19. That PR was merged the 16th at 11am; [the first failure](https://jenkins.cilium.io/job/cilium-master-k8s-1.16-kernel-4.19/4127/) was the same day at 5pm. cc @brb

**Comment by brb** [comment]
Some example failure taken from the sysdump in the issue description:
```
time="2023-02-16T17:43:30Z" level=error msg="Error executing local command 'kubectl exec -n kube-system log-gatherer-f5dbg -- dig +short app1-service.default.svc.cluster.local @10.96.0.10'" error="exit status 9"
cmd: "kubectl exec -n kube-system log-gatherer-f5dbg -- dig +short app1-service.default.svc.cluster.local @10.96.0.10" exitCode: 9 duration: 15.11169609s stdout:
;; connection timed out; no servers could be reached
```

As the log-gatherer pod is running in the host netns, the service xlation should be done by kube-proxy's iptables rules.

Meanwhile, kube-proxy pods have plenty of:
```
2023-02-16T17:01:55.895226962Z E0216 17:01:55.895129       1 healthcheck.go:169] node k8s2 failed to start healthcheck "default/test-lb-local-k8s2" on port 31760: listen tcp :31760: bind: address already in use
```

It might be red-herring, but the above happens because of ` --enable-health-check-nodeport='true'"` in the agent. The iptables rules exist though.

Anyway, I don't see any requests from the log-gatherer-f5dbg to the CoreDNS service in the hubble flow logs. Also, the relevant iptables counters are zero.

**Comment by brb** [comment]
Ah, I think the culprit is the following:
```
> cat bpftool-cgroup-tree--run-cilium-cgroupv2.md
CgroupPath
ID       AttachType      AttachFlags     Name
/run/cilium/cgroupv2
89477    cgroup_inet4_connect                 cil_sock4_conne
89461    cgroup_inet6_connect                 cil_sock6_conne
89481    cgroup_inet4_post_bind                 cil_sock4_post_
89465    cgroup_inet6_post_bind                 cil_sock6_post_
89485    cgroup_udp4_sendmsg                 cil_sock4_sendm
89469    cgroup_udp6_sendmsg                 cil_sock6_sendm
89489    cgroup_udp4_recvmsg                 cil_sock4_recvm
89473    cgroup_udp6_recvmsg                 cil_sock6_recvm
```
The Cilium's cgroup progs (socket LB) have not been unloaded from a previous run.

**Comment by brb** [comment]
> Anyway, a temporary fix for that 4.19 job is to figure out which test case has accidentally enabled the socket LB

The offender - `K8sPolicyTestExtended[0m [90mValidate toEntities KubeAPIServer` :police_officer:
